blog-duplicator-cli.php

```diff
@@ -48,79 +57,136 @@ function __invoke( $args, $assoc_args ) {
+		$src_baseurl       = $src_wp_upload_dir['baseurl'];
 
-		// duplicate tables
+		// duplicate tables.
 		switch_to_blog( $id );
-			// make upload destination
-			$dest_wp_upload_dir = wp_upload_dir();
-			$dest_basedir = $dest_wp_upload_dir['basedir'];
-			$dest_baseurl = $dest_wp_upload_dir['baseurl'];
-			wp_mkdir_p( $dest_basedir );
-
-			// copy files
-			WP_CLI::line( "Duplicating uploads..." );
-			$this->verbose_line( 'Running command:', "rsync -a {$src_basedir}/ {$dest_basedir} --exclude sites", $verbose );
-			shell_exec( "rsync -a {$src_basedir}/ {$dest_basedir} --exclude sites" );
-
-			// duplicate tables
-			$url = home_url();
-			WP_CLI::line( "Duplicating tables..." );
-			foreach ( $wpdb->tables('blog') as $k => $table ) {
-				$origin_table = $origin_tables[ $k ];
-				$wpdb->query( "TRUNCATE TABLE $table" );
-				$wpdb->query( "INSERT INTO $table SELECT * FROM $origin_table" );
+
+		// make upload destination.
+		$dest_wp_upload_dir = wp_upload_dir();
+		$dest_basedir       = $dest_wp_upload_dir['basedir'];
+		$dest_baseurl       = $dest_wp_upload_dir['baseurl'];
+		wp_mkdir_p( $dest_basedir );
+
+		// copy files.
+		if ( ! $skip_copy_files ) {
+			$is_shell_exec_enabled = is_callable( 'shell_exec' ) && false === stripos( ini_get( 'disable_functions' ), 'shell_exec' );
+
+			if ( ! $is_shell_exec_enabled ) {
+				WP_CLI::warning( 'shell_exec is disabled, skipping file copying!' );
+			} else {
+				$is_rsync_installed = ! empty( shell_exec( 'which rsync' ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.system_calls_shell_exec
+
+				if ( $is_rsync_installed ) {
+					WP_CLI::line( 'Duplicating uploads...' );
+					$this->verbose_line( 'Running command:', "rsync -a {$src_basedir}/ {$dest_basedir} --exclude sites", $verbose );
+					shell_exec( // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.system_calls_shell_exec
+						sprintf(
+							'rsync -a %s/ %s --exclude sites',
+							escapeshellarg( $src_basedir ),
+							escapeshellarg( $dest_basedir )
+						)
+					);
+				} else {
+					WP_CLI::warning( 'Cannot find rsync, skipping file copying!' );
+				}
+			}
+		} else {
+			WP_CLI::warning( 'SKIPPING Duplicating uploads...' );
+		}
+
+		// duplicate tables.
+		// phpcs:disable WordPress.DB.DirectDatabaseQuery,WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		$url = home_url();
+		WP_CLI::line( 'Duplicating tables...' );
+
+		$extra_tables = array();
+		foreach ( apply_filters( 'blog_duplicator_extra_tables', array() ) as $extra_table ) {
+			$extra_tables[ $extra_table ] = $wpdb->prefix . $extra_table;
+		}
+
+		$blog_tables = array_merge( $extra_tables, $wpdb->tables( 'blog' ) );
+		foreach ( $blog_tables as $k => $table ) {
+			$origin_table = $origin_tables[ $k ];
+
+			$this->verbose_line( 'Running SQL:', "TRUNCATE TABLE $table", $verbose );
+			$wpdb->query( "TRUNCATE TABLE $table" );
+
+			// Remove blocked options from option table before import.
+			if ( $wpdb->prefix . 'options' === $table ) {
```

Comment:
I think `$wpdb->prefix . 'options'` can be simplified down to `$wpdb->options`

Reply:
🤦